pkg/scraper/client/resource/decode_test.go

```diff
@@ -0,0 +1,136 @@
+			Value:     700,
+			Timestamp: model.Time(scrapeTime.Add(30*time.Millisecond).Unix() / 1e6),
+		}
+		sample8 := model.Sample{Metric: model.Metric{model.MetricNameLabel: "container_memory_working_set_bytes", "container": "container1", "namespace": "ns1", "pod": "pod2"},
+			Value:     800,
+			Timestamp: model.Time(scrapeTime.Add(30*time.Millisecond).Unix() / 1e6),
+		}
+		sample9 := model.Sample{Metric: model.Metric{model.MetricNameLabel: "container_cpu_usage_seconds_total", "container": "container1", "namespace": "ns2", "pod": "pod1"},
+			Value:     900,
+			Timestamp: model.Time(scrapeTime.Add(40*time.Millisecond).Unix() / 1e6),
+		}
+		sample10 := model.Sample{Metric: model.Metric{model.MetricNameLabel: "container_memory_working_set_bytes", "container": "container1", "namespace": "ns2", "pod": "pod1"},
+			Value:     1000,
+			Timestamp: model.Time(scrapeTime.Add(40*time.Millisecond).Unix() / 1e6),
+		}
+		sample11 := model.Sample{Metric: model.Metric{model.MetricNameLabel: "container_cpu_usage_seconds_total", "container": "container1", "namespace": "ns3", "pod": "pod1"},
+			Value:     1100,
+			Timestamp: model.Time(scrapeTime.Add(50*time.Millisecond).Unix() / 1e6),
+		}
+		sample12 := model.Sample{Metric: model.Metric{model.MetricNameLabel: "container_memory_working_set_bytes", "container": "container1", "namespace": "ns3", "pod": "pod1"},
+			Value:     1200,
+			Timestamp: model.Time(scrapeTime.Add(50*time.Millisecond).Unix() / 1e6),
+		}
+		samples = []*model.Sample{}
+		samples = append(samples, &sample1, &sample2, &sample3, &sample4, &sample5, &sample6, &sample7, &sample8, &sample9, &sample10, &sample11, &sample12)
+	})
+
+	It("should use the decode time from the CPU", func() {
+		By("removing some times from the data")
+
+		By("decoding")
+		batch := decodeBatch(samples, "node1")
+
+		By("verifying that the scrape time is as expected")
+		Expect(batch.Nodes["node1"].Timestamp).To(Equal(time.Unix(0, int64(samples[0].Timestamp*1e6))))
+		Expect(batch.Pods[apitypes.NamespacedName{Namespace: "ns1", Name: "pod1"}].Containers["container1"].Timestamp).To(Equal(time.Unix(0, int64(samples[2].Timestamp*1e6))))
+		Expect(batch.Pods[apitypes.NamespacedName{Namespace: "ns1", Name: "pod2"}].Containers["container1"].Timestamp).To(Equal(time.Unix(0, int64(samples[6].Timestamp*1e6))))
+	})
+
+	It("should continue on missing CPU or memory metrics", func() {
+		By("removing some data from the raw samples")
+		samples[1].Timestamp = 0
+		samples[1].Value = 0
+		samples[4].Timestamp = 0
+		samples[4].Value = 0
+		samples[6].Value = 0
+		samples[9].Timestamp = 0
+		samples[9].Value = 0
+		samples[11].Value = 0
+
+		By("decoding")
+		batch := decodeBatch(samples, "node1")
+
+		By("verifying that the batch has all the data, save for what was missing")
+		Expect(batch.Pods).To(HaveLen(0))
+		Expect(batch.Nodes).To(HaveLen(0))
+	})
+
+	It("should skip on cumulative CPU equal zero", func() {
```

Comment:
Can you use pkg/scraper/client/summary/decode_test.go for this?

Reply:
Let me see if there is a difference between them.

Reply:
@serathius, I have fixed it.

Reply:
Could you try to modify the PR so we can see the diff between old decode_test.go and new one? Same what you did for client.